ember-client/src/main/scala/org/http4s/ember/client/internal/ClientHelpers.scala

```diff
@@ -191,4 +196,31 @@ private[client] object ClientHelpers {
               new java.net.SocketException("Fresh connection from pool was not open")))
         )
     }
+
+  private[ember] object RetryLogic {
+    import org.http4s.client.middleware._
+    import org.http4s.syntax.all._
+    // import org.http4s.headers.`Idempotency-Key`
+
+    private val retryNow = 0.seconds.some
+    def retryUntilFresh[F[_]]: RetryPolicy[F] = { (req, result, retries) =>
+      if (emberDeadFromPoolPolicy(req, result) && retries <= 2) retryNow
```

Comment:
Should this be `retries < 2`? I thought @rossabaker said only retry once. Won't this retry twice?

Reply:
I believe this is 1 indexed. 

https://github.com/http4s/http4s/blob/main/client/shared/src/main/scala/org/http4s/client/middleware/Retry.scala#L99